trace/header.go

```diff
@@ -1,34 +1,60 @@
 package trace
 
 import (
+	"log"
 	"net/http"
-
-	"github.com/twinj/uuid"
 )
 
 const (
-	HeaderId = "x-request-id"
-
-	HeaderParentId = "x-parent-id"
+	// HeaderSpanID is the header name in requests for the SpanID.
+	HeaderSpanID = "Trace-Span"
+	// HeaderTraceID is the header name in requests for the TraceID.
+	HeaderTraceID = "Trace-Trace"
 )
 
-func GetId(r *http.Request) string {
-	return r.Header.Get(HeaderId)
+// Generates a new SpanID from a request. If there is no span information in the request, a root SpanID will be generated.
+func NewSpanIDFromRequest(req *http.Request) SpanID {
+	parentSpanID := ParseSpanID(req)
+	if parentSpanID.Empty() {
+		return NewRootSpanID()
+	}
+	return NewSpanID(parentSpanID)
+}
+
+// Returns the SpanID contained in the request.
+func ParseSpanID(req *http.Request) SpanID {
+	spanIDStr := req.Header.Get(HeaderSpanID)
+	traceIDStr := req.Header.Get(HeaderTraceID)
+	spanID, err := ParseID(spanIDStr)
+	if err != nil {
+		log.Println("Failed to parse spanID string value, using 0.")
```

Comment:
I think logging in a library is very annoying. Especially if you don't provide any control over it. What if I want to use a different logger. What I've seen recently is to create an interface that matches the stdlib's logger and make it settable. The default could be an instance of the stdlib's logger. [Something like this](https://github.com/Shopify/sarama/blob/7cdfc5b3e054fe72ad79af2a2984bbffcfcf3c64/sarama.go#L23-L30).

Also if you ignore the error please be explicit about the fallback `spanID`. This could hide some nasty bugs if the the return value of `ParseID` changes in the error case. Also this introduces some tight coupling.